oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs

```diff
@@ -548,64 +570,80 @@ impl ManifestStorage {
                         .bind::<diesel::sql_types::BigInt, _>(manifest.base_version() as i64)
                         .bind::<diesel::sql_types::BigInt, _>(manifest.base_version() as i64)
                         .bind::<diesel::sql_types::Binary, _>(vlob_id)
-                        .execute(conn))?;
-
-            for pending_chunk_ids_chunk in pending_chunk_ids.chunks(SQLITE_MAX_VARIABLE_NUMBER) {
-                conn.exec(|conn| {
-                    diesel::delete(
-                        chunks::table.filter(chunks::chunk_id.eq_any(pending_chunk_ids_chunk)),
-                    )
-                    .execute(conn)
-                })?;
+                        .execute(conn)
+            }).await?;
+
+            for pending_chunk_id in chunk_ids
+                .chunks(LOCAL_DATABASE_MAX_VARIABLE_NUMBER)
+                .map(|chunk| chunk.to_vec())
+            {
+                self.conn
+                    .exec(move |conn| {
+                        diesel::delete(
+                            chunks::table.filter(chunks::chunk_id.eq_any(pending_chunk_id)),
```

Comment:
use `pending_chunk_id.iter().map(ChunkOrBlockID::as_ref)` there, which replace the `map` and avoid the `.to_vec()`